tests/gtest/cppapi_test.cpp

```diff
@@ -0,0 +1,104 @@
+#include <map>
+
+#include "ctypes.h"
+#include "cppapi.hpp"
+
+#include "gtest/gtest.h"
+#include "gmock/gmock.h"
+
+namespace
+{
+    const std::string SAMPLES_10 = "file://10_samples_default.vds";
+    const std::string CREDENTIALS = "";
+
+    std::map<float, std::vector<float>> samples_10_data(float fill)
+    {
+        std::map<float, std::vector<float>> points;
+        points[110] = {-4.5, -3.5, -2.5, -1.5, -0.5, 0.5, 1.5, 2.5, 3.5, 4.5};
+        points[111] = {4.5, 3.5, 2.5, 1.5, 0.5, -0.5, -1.5, -2.5, -3.5, -4.5};
+        points[310] = {25.5, -14.5, -12.5, -10.5, -8.5, -6.5, -4.5, -2.5, -0.5, 25.5};
+        points[311] = {25.5, 0.5, 2.5, 4.5, 6.5, 8.5, 10.5, 12.5, 14.5, 25.5};
+        points[510] = {25.5, 4.5, 8.5, 12.5, 16.5, 20.5, 24.5, 20.5, 16.5, 8.5};
+        points[511] = {25.5, -4.5, -8.5, -12.5, -16.5, -20.5, -24.5, -20.5, -16.5, -8.5};
+        points[fill] = {fill};
+        return points;
+    }
```

Comment:
It helps, but I still think it's messy and unnecessary and that using a 3D array would be clearer (because that is the physical shape the data). But we can agree to disagree 😄